src/RegExLib.Web/Endpoints/Expressions/Create.CreateExpressionResult.cs

```diff
@@ -0,0 +1,9 @@
+﻿using RegExLib.Web.ApiModels;
+
+namespace RegExLib.Web.Endpoints.Expressions
+{
+  public class CreateExpressionResult : CreateExpressionCommand
```

Comment:
Having a Result derive from a Command seems like a smell to me. I understand why you're doing it, but from an OOAD perspective, it means that a Result _IS-A_ Command which is.. smelly 🤷 

Also, I think "Request/Response" might be a better naming convention than "Command/Result" for API Endpoints since **endpoints** handle requests, (whereas **command handlers** handle commands).

Reply:
Result IS-A Command yes it is true because both will carry Expression. maybe naming needs to change.
We need to change here samples here also [ardalis.ApiEndpoints](https://github.com/ardalis/ApiEndpoints) 

> 
> 
> Having a Result derive from a Command seems like a smell to me. I understand why you're doing it, but from an OOAD perspective, it means that a Result _IS-A_ Command which is.. smelly 🤷
> 
> Also, I think "Request/Response" might be a better naming convention than "Command/Result" for API Endpoints since **endpoints** handle requests, (whereas **command handlers** handle commands).

Reply:
>Result IS-A Command yes it is true because both will carry Expression.

Even though they might contain the same data, they represent two different things _conceptually._

> maybe naming needs to change. We need to change the samples here also ardalis.ApiEndpoints

Yeah, the naming there is inconsistent with the [ToDoItems stuff from CleanArchitecture](https://github.com/ardalis/CleanArchitecture/tree/master/src/CleanArchitecture.Web/Endpoints/ToDoItems). Deferring to @ardalis on that

Reply:
Done in [commit 36854](https://github.com/ardalis/RegExLib.com/pull/44/commits/36854189abe5ebd0f65120af4bfdeaf0890eda6c)

Reply:
> Having a Result derive from a Command seems like a smell to me. I understand why you're doing it, but from an OOAD perspective, it means that a Result _IS-A_ Command which is.. smelly 🤷 
> 
> 
> 
> Also, I think "Request/Response" might be a better naming convention than "Command/Result" for API Endpoints since **endpoints** handle requests, (whereas **command handlers** handle commands).

Thinking about this more in the context of CQS, there _is_ an advantage to the explicit naming of “FooCommand” vs “FooQuery” - I know that the command will have side effects whereas the query should not. Also, I’ve seen implementations of “command” handlers often handle “queries” as well as just “commands” as a matter of pragmatism (and DRY) 🤔